### PR TITLE
task: also instrument streams

### LINKED ISSUE #30: Allow `Stream`s to be instrumented
Implement `Instrumented<T>` for `T: Stream`.

### PR DESCRIPTION
**Pull request by duarten**
Using pretty much the same logic as for futures, instrument streams as
well.

Fixes #30

### CODE REVIEW

File: Cargo.toml [hunk 1]
```diff
@@ -19,6 +19,7 @@ default = ["rt"]
 rt = ["tokio"]
 
 [dependencies]
+futures-core = "0.3.25"
```
Comment (LucioFranco) [inline Cargo.toml:22]: Could prob depend on `tokio-stream` instead?

Comment (duarten) [reply Cargo.toml:22]: Will change!

**Comment by Darksonn** [comment]
You will need to merge in master once #32 is merged to fix the CI failures.

**Comment by Darksonn** [comment]
You can merge master now.

**Comment by duarten** [comment]
Ping 😅

**Comment by duarten** [comment]
The doc tests also fail for me on main.

**Comment by jswrenn** [comment]
Since the crate monitors tokio's internals, there's an inherent amount of flakiness. Perhaps we could use [nextest with retries](https://nexte.st/book/retries.html) in CI instead.

**Comment by Darksonn** [comment]
Is it just a flaky test? I saw it fail several times in a row.

**Review by Darksonn — APPROVED** [review]
Anyway, the test failure isn't your fault, and would need to be fixed in a separate PR regardless. I'm going to merge this now.

**Pull request merged**